internal/dag/builder_test.go

```diff
@@ -28,8 +30,375 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
 )
 
+func TestDAGInsertServiceAPIs(t *testing.T) {
```

Comment:
Worth adding tests (or maybe a tracking issue for adding tests) around the interactions between different resources (HTTPProxy, Ingress, Gateway APIs etc.)

Reply:
👍  https://github.com/projectcontour/contour/issues/3367